.infrastructure/2_ecr.tf

```diff
@@ -0,0 +1,31 @@
+resource "aws_ecr_repository" "image-repository" {
+  name = var.ecr_repository
+
+  image_scanning_configuration {
+    scan_on_push = true
+  }
+}
+
+resource "aws_ecr_lifecycle_policy" "policy" {
+  repository = aws_ecr_repository.image-repository.name
+
+  policy = <<EOF
+  {
+      "rules": [
+          {
+              "rulePriority": 1,
+              "description": "Keep last 5 images",
+              "selection": {
+                  "tagStatus": "tagged",
+                  "tagPrefixList": ["v"],
+                  "countType": "imageCountMoreThan",
+                  "countNumber": 5
+              },
+              "action": {
+                  "type": "expire"
+              }
+          }
+      ]
+  }
+  EOF
+}
```

Comment:
Why did you choose `ecr` and not [GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#about-the-container-registry)?

Reply:
It is possible to use GitHub Container Registry or Docker Hub, but ECR is a native AWS service and therefore better integrated with Cluster ECS Fargate and related task definition. It also allows automatic pulling of Fargate every time the image is updated. For ECR, present in the relevant RFC, the first 50 GB of stored data are free as is internet traffic.

Reply:
The same is true between **GitHub Action** and **GitHub Container Registry**, they are better integrated. We don't know where the container image is going to be uploaded yet.

Reply:
This is also true, so I would say to keep both options (ECR and GitHub Container Registry) viable/active